src/admin/index.js

```diff
@@ -87,9 +99,123 @@ module.exports = ({ retry = { retries: 5 }, logger: rootLogger, cluster }) => {
     })
   }
 
+  /**
+   * @param {string} groupId
+   * @param {string} topic
+   * @return {Promise}
+   */
+  const fetchOffsets = async ({ groupId, topic }) => {
+    if (!groupId) {
+      throw new KafkaJSNonRetriableError(`Invalid groupId ${groupId}`)
+    }
+
+    if (!topic) {
+      throw new KafkaJSNonRetriableError(`Invalid topic ${topic}`)
+    }
+
+    const partitions = await findTopicPartitions(cluster, topic)
+    const coordinator = await cluster.findGroupCoordinator({ groupId })
+    const partitionsToFetch = partitions.map(partition => ({ partition }))
+
+    const { responses } = await coordinator.offsetFetch({
+      groupId,
+      topics: [{ topic, partitions: partitionsToFetch }],
+    })
+
+    return responses
+      .filter(response => response.topic === topic)
```

Comment:
If the coordinator is unaware of the topic, what will we return here? I assume undefined, but maybe we should rather reject? Or will the offsetFetch already reject?

Reply:
Good point, if the `metadata` request can't create the topic it might throw an error. I will investigate

Reply:
It will throw `UNKNOWN_TOPIC_OR_PARTITION`

```javascript
{ 
  KafkaJSProtocolError: This server does not host this topic-partition
    at createErrorFromCode (./src/protocol/error.js:463:10)
    at Object.parse (./src/protocol/requests/metadata/v0/response.js:56:11)
    at Connection.send (./src/network/connection.js:249:35)
    at process._tickCallback (internal/process/next_tick.js:68:7)
  name: 'KafkaJSProtocolError',
  retriable: true,
  type: 'UNKNOWN_TOPIC_OR_PARTITION',
  code: 3
}
```